airbyte-integrations/bases/base-java/src/main/resources/ssh-tunnel-spec.json

```diff
@@ -2,6 +2,7 @@
   "type": "object",
   "title": "SSH Tunnel Method",
   "description": "Whether to initiate an SSH tunnel before connecting to the database, and if so, which kind of authentication to use.",
+  "group": "security",
```

Comment:
Unfortunately the ssh tunnel spec field is in this separate JSON file and gets copied into all of the JDBC source connectors.

Adding the `group` here is fine for postgres because as part of this PR, this group is also set on the other security fields. But this will be a problem for the non-postgres connectors that don't have any groups set, as this will cause the SSH tunnel method to go into its own group. 

Maybe the best approach here is to also migrate the other JDBC connectors to have groups like we did with postgres here? Open to other suggestions though

Reply:
This is a little annoying because we would need to make sure the groups are set completely, otherwise things will look weird (as named groups are sorted to the top).

This is how the injection is happening for postgres:
https://github.com/airbytehq/airbyte/blob/009940404cd90e72cb42f6b2ab91549ced54e7d5/airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java#L118

https://github.com/airbytehq/airbyte/blob/2e099acc52aeddc172c5ee66ed98426c69449a4a/airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/ssh/SshHelpers.java#L23

https://github.com/airbytehq/airbyte/blob/2e099acc52aeddc172c5ee66ed98426c69449a4a/airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/ssh/SshWrappedSource.java#L37

Maybe we can parameterize `SshWrappedSource` and `injectSshIntoSpec` with an optional resource file name and create `grouped-ssh-tunnel-spec.json` and use it for postgres specifically?

Reply:
@flash1293 I went mostly with your suggestion, but I wanted to avoid duplicating the `ssh-tunnel-spec.json` just to add a single `group` field, so I instead updated these classes to parametrize those methods to take in an optional `sshGroup` string which is set on the tunnel_method spec if present before it is set on the connector spec. I used overloaded methods here so other connectors should not need any changes.

See the commit of these changes here: https://github.com/airbytehq/airbyte/pull/24127/commits/b0404a1c5ca5693e9b1943ed0aacc04a05cd3ab1